crates/bevy_image/src/serialized_image.rs

```diff
@@ -0,0 +1,177 @@
+            aspect: descriptor.aspect,
+            base_mip_level: descriptor.base_mip_level,
+            mip_level_count: descriptor.mip_level_count,
+            base_array_layer: descriptor.base_array_layer,
+            array_layer_count: descriptor.array_layer_count,
+        }
+    }
+
+    fn into_texture_view_descriptor(self) -> TextureViewDescriptor<Option<&'static str>> {
+        TextureViewDescriptor {
+            // Not used for asset-based images other than debugging
+            label: None,
+            format: self.format,
+            dimension: self.dimension,
+            usage: self.usage,
+            aspect: self.aspect,
+            base_mip_level: self.base_mip_level,
+            mip_level_count: self.mip_level_count,
+            base_array_layer: self.base_array_layer,
+            array_layer_count: self.array_layer_count,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+enum SerializedTextureDataOrder {
+    LayerMajor,
+    MipMajor,
+}
+
+impl SerializedTextureDataOrder {
+    fn from_texture_data_order(order: TextureDataOrder) -> Self {
+        match order {
+            TextureDataOrder::LayerMajor => SerializedTextureDataOrder::LayerMajor,
+            TextureDataOrder::MipMajor => SerializedTextureDataOrder::MipMajor,
+        }
+    }
+
+    fn into_texture_data_order(self) -> TextureDataOrder {
+        match self {
+            SerializedTextureDataOrder::LayerMajor => TextureDataOrder::LayerMajor,
+            SerializedTextureDataOrder::MipMajor => TextureDataOrder::MipMajor,
+        }
+    }
+}
+
+impl SerializedImage {
+    /// Creates a new [`SerializedImage`] from an [`Image`].
+    pub fn from_image(image: Image) -> Self {
+        Self {
+            data: image.data,
+            data_order: SerializedTextureDataOrder::from_texture_data_order(image.data_order),
+            texture_descriptor: TextureDescriptor {
+                label: (),
+                size: image.texture_descriptor.size,
+                mip_level_count: image.texture_descriptor.mip_level_count,
+                sample_count: image.texture_descriptor.sample_count,
+                dimension: image.texture_descriptor.dimension,
+                format: image.texture_descriptor.format,
+                usage: image.texture_descriptor.usage,
+                view_formats: (),
+            },
+            sampler: image.sampler,
+            texture_view_descriptor: image.texture_view_descriptor.map(|descriptor| {
+                SerializedTextureViewDescriptor::from_texture_view_descriptor(descriptor)
+            }),
+        }
+    }
+
+    /// Create an [`Image`] from a [`SerializedImage`].
+    pub fn into_image(self) -> Image {
+        Image {
+            data: self.data,
+            data_order: self.data_order.into_texture_data_order(),
+            texture_descriptor: TextureDescriptor {
+                // Not used for asset-based images other than debugging
```

Comment:
Can you please add doc comments to this field on Image explaining what it's actually for? I presume they weren't good enough, which is why you needed to ask on Discord :p